### ISSUE TITLE
RED Unable to start XML-RPC

### ISSUE THREAD

**Issue by Mike2206** [nokia/RED#295]
Hi,

I know this topic has been here before but none of the answers there helped me and since the topics are closed already I have decided to make a new one. 

I have this "Unable to start XML-RPC server on file" error when trying to discover installed frameworks. When trying to manually set the folder for Python with RF installed it will leave the table blank for some reason. 

I'm running a Windows 7 32-bit machine. Tried to reinstall RED and nothing really helps. Also tried to run Eclipse as Admin.

Any help will be appreciated. 

Here is my log:
[Log.log](https://github.com/nokia/RED/files/3101056/Log.log)

**Comment by modrz** [comment]
Hi,

Exception from you log file may occur in 2 cases:
1. Java/OS returns **false** when checking if python executable file exists under found path
2. There is a problem with **XmlRpc** client/server connection

Let us check if you problem is related to java File::exists.

Execute following jar file from the attachment:
`java -jar InterpreterPathTest.jar`

It should return something like this:
```
INTERPRETER:           Python
INTERPRETER DIR:       C:\Python27
INTERPRETER PATH:      C:\Python27\python.exe
PATH EXISTS:           true
ABSOLUTE PATH EXISTS:  true

INTERPRETER:           Jython
INTERPRETER DIR:       C:\jython2.7.1\bin
INTERPRETER PATH:      C:\jython2.7.1\bin\jython.exe
PATH EXISTS:           true
ABSOLUTE PATH EXISTS:  true

```

[InterpreterPathTest.zip](https://github.com/nokia/RED/files/3106314/InterpreterPathTest.zip)

**Comment by michalanglart** [comment]
Hi,

I was able to reproduce this issue when my temporary files folder is located under path containing non-ASCII character ś (s + accute) - same as yours (michaś). We need to have a look. 

As a workaround you may either change the environment variables in single console session:
`
set TEMP=c:/other/location/
set TMP=c:/other/location/
RED.exe
`

or you may change it globally however this would influence all the other applications in your system.

**Comment by Mike2206** [comment]
Hey,

Thx for your answers. In the attachment you got a screenshot of my console after executing the .jar file. 
![jar](https://user-images.githubusercontent.com/49831799/56610721-6914ee00-6610-11e9-8cbd-ad17503870b9.png)

I have installed the newest Java and I have the JDK installed on my system as well.

When I try "Discover" option in RED it does not recognize the special character. But I have reinstalled Python to another location that does not contain any special characters. Now when I try to go "Discover" it takes way longer before throwing the same exception but without the incorrect path so I'm not sure if this is related to the location of Python.
![XML-RPC](https://user-images.githubusercontent.com/49831799/56612784-7bddf180-6615-11e9-9b7e-21dd4c23e592.png)

Also for some reason the command line you have provided does not change the TEMP location, at least I dont get any confirmation. I might do something wrong tho, I'm a bit newbie to all of this so please tell me step by step what to do :)

**Comment by modrz** [comment]
Hi,
You can check if environment variable was changed using following commands in console
```
echo %TEMP%
echo %TMP%
set TEMP=c:/other/location/
set TMP=c:/other/location/
echo %TEMP%
echo %TMP%
```
Then you have to start RED.exe in the same console session.

Your issue is caused by Python 2 problems with utf-8 encoding.
You can test it with simple "Helo World" python file:
- place it in some directory with unicode characters in path, for example your home directory
- run following command in console: `python C:\Users\Michaś\some_python_file.py`
- script running fails with Python 2 and works with Python 3

Same problem occurs in RED, when python code is run. Workaround for now is changing TEMP path to some path without unicode character.

**Comment by Mike2206** [comment]
Ok. I can change the TEMP directory but how am I supposed to start RED.exe when I don't use a stand alone RED? i'm using RED via Eclipse since I'm using a 32-bit system

I was able to run a Hello World python file in my console 
![hello](https://user-images.githubusercontent.com/49831799/56849094-8c7db880-68f0-11e9-999f-1a4c2c754f03.png)

The problem is that after changing my TEMP folder to a different location (in this case D:\TEMP I should run RED.exe and I can't do that on a 32-bit system right?

**Comment by michalanglart** [comment]
If you're using eclipse instead of RED then your installation should have `eclipse.exe` which is used to start it.

**Comment by Mike2206** [comment]
Hi guys I just wanted to say that the solution provided doesn't work - Eclipse will still produce the same error! I have installed Python in a folder with no special signs as well as I did with and it is still giving me the same response :(

**Comment by michalanglart** [comment]
Hi,

You've already shown us here: https://github.com/nokia/RED/issues/295#issuecomment-485960553 that you had python installed under location without problematic characters.

Please read carefully the comment above from @modrz - it states that the problem occurs when the script executed by python is located under such path, not the interpreter itself. Have you tried changing your TEMP location in a manner @modrz suggested?

**Comment by michalanglart** [comment]
We have some fix which works in my case so should probably work for you too. We plan to release next version on monday next week, so please have a look then.

**Comment by michalanglart** [comment]
Please check with 0.8.12 if newest changes fixes it on your side (it fixed the problem on my side)

**Comment by Mike2206** [comment]
Hi guys,

I have installed the latest update and I have an error again (screenshot attached)
![Error](https://user-images.githubusercontent.com/49831799/58381230-d4106680-7fba-11e9-9682-6a4ce5aa3ec1.png)

I have tried changing the Temp folder to default or to a different path, it produces the same error message no matter where the temp folder is located.

**Comment by kenifanying** [comment]
hi,

I have also encounterd into this problem after upgrade 0.8.12,

**Comment by michalanglart** [comment]
Can you both attach your `.log` file from `workspace/metadata`?

**Comment by kenifanying** [comment]
This is my log file,

[log.txt](https://github.com/nokia/RED/files/3222284/log.txt)

**Comment by Mike2206** [comment]
Here is mine:

[log.log](https://github.com/nokia/RED/files/3222336/log.log)

**Comment by michalanglart** [comment]
@Mike2206 the newest entry in your `.log` file has a timestamp 2019-04-27 so obviously almost a month before latest release. Have you updated to RED 0.8.12?

**Comment by Mike2206** [comment]
Yes i have updated it, i will look into it again when i get back from work but im very sure that I installed the upgrade

**Comment by michalanglart** [comment]
@Mike2206 Ok, please give us the newest `.log` file then

@kenifanying Your `.log` file has also other exceptions suggesting there is a connectivity issue through HTTP (exceptions thrown when your eclipse was looking for updates). Is there anything specific in your network configuration? Proxies, firewalls, blocked ports?

**Comment by kenifanying** [comment]
> 
> 
> @Mike2206 Ok, please give us the newest `.log` file then
> 
> @kenifanying Your `.log` file has also other exceptions suggesting there is a connectivity issue through HTTP (exceptions thrown when your eclipse was looking for updates). Is there anything specific in your network configuration? Proxies, firewalls, blocked ports?

I don't have any special configuration in my network, and it also occured when i try to reload library.

**Comment by michalanglart** [comment]
@kenifanying Can you please try running server manually and connecting RED to it, or displaying servers console? It is described here: http://nokia.github.io/RED/help/user_guide/diagnostics.html

**Comment by kenifanying** [comment]
> 
> 
> @kenifanying Can you please try running server manually and connecting RED to it, or displaying servers console? It is described here: http://nokia.github.io/RED/help/user_guide/diagnostics.html

It seems that no strange output,

![snipaste_20190527_184347](https://user-images.githubusercontent.com/4479181/58414910-66923200-80af-11e9-9297-874e5c17bb3a.png)

[debug1.txt](https://github.com/nokia/RED/files/3223410/debug1.txt)

**Comment by michalanglart** [comment]
Actually it looks like the server just crashes when generating library specification file for `OnlineUserMngSMP` library which you've also reported here: #281 

Can you please run session server manually (`-Dred.connectToServerAt=xxx`)instead of showing console output and look what is happening with server? Does it crash with more info than here?

**Comment by kenifanying** [comment]
> 
> 
> Actually it looks like the server just crashes when generating library specification file for `OnlineUserMngSMP` library which you've also reported here: #281
> 
> Can you please run session server manually (`-Dred.connectToServerAt=xxx`)instead of showing console output and look what is happening with server? Does it crash with more info than here?

I can't reproduce this problem by running session server manually (`-Dred.connectToServerAt=127.0.0.1:12345`)instead of showing console output. But I keep get this error:
```
Library specification generation problem

Problem occurred during generating libdoc for 'OnlineUserMngSMP' library located at 'C:\Users\tony\robotframework\rf_pi\lib\WebLibrary\PageObject\OnlineUserMngSMP.py'.
RED python session problem. Following exception has been thrown by python service:
            Traceback (most recent call last):
              File "robot_session_server.py", line 42, in inner
                result['result'] = func(*args, **kwargs)
              File "robot_session_server.py", line 91, in inner
                return to_call(*args, **kwargs)
              File "robot_session_server.py", line 124, in inner
                return to_call(*args, **kwargs)
              File "robot_session_server.py", line 64, in inner
                return func(*args, **kwargs)
              File "robot_session_server.py", line 323, in create_libdoc
                return red_libraries.create_libdoc(libname, format)
              File "C:\Users\tony\robot\red_libraries.py", line 67, in create_libdoc
                raise Exception(result)
            Exception: Importing test library 'C:\Users\tony\robotframework\rf_pi\lib\WebLibrary\PageObject\OnlineUserMngSMP.py' failed: implement_array_function method already has a docstring
            Traceback (most recent call last):
              File "C:\Users\tony\robotframework\rf_pi\lib\WebLibrary\PageObject\OnlineUserMngSMP.py", line 9, in <module>
                from lib.WebLibrary.BaseObject.BaseObject import *
              File "C:\Users\tony\robotframework\rf_pi\lib\WebLibrary\BaseObject\BaseObject.py", line 11, in <module>
                from PIL import Image, ImageEnhance
              File "C:\Python27\lib\site-packages\PIL\ImageEnhance.py", line 21, in <module>
                from . import Image, ImageFilter, ImageStat
              File "C:\Python27\lib\site-packages\PIL\ImageFilter.py", line 23, in <module>
                import numpy
              File "C:\Python27\lib\site-packages\numpy\__init__.py", line 142, in <module>
                from . import core
              File "C:\Python27\lib\site-packages\numpy\core\__init__.py", line 40, in <module>
                from . import multiarray
              File "C:\Python27\lib\site-packages\numpy\core\multiarray.py", line 12, in <module>
                from . import overrides
              File "C:\Python27\lib\site-packages\numpy\core\overrides.py", line 46, in <module>
                """)
            PYTHONPATH:
              C:\Users\tony\robot\lib\Tcl\lib\SpirentHltApi\SourceCode\hltapiForPython\
              C:/Users/tony/robotframework/rf_pi/lib/WebLibrary/PageObject
              C:\Users\tony\robotframework\rf_pi\lib\Tcl\lib\SpirentHltApi\SourceCode\hltapiForPython
              C:\Users\tony\robotframework\rf_pi
              C:\Users\tony\robot
              C:\Windows\system32\python27.zip
              C:\Python27\DLLs
              C:\Python27\lib
              C:\Python27\lib\plat-win
              C:\Python27\lib\lib-tk
              C:\Python27
              C:\Python27\lib\site-packages
              C:\Python27\lib\site-packages\win32
              C:\Python27\lib\site-packages\win32\lib
              C:\Python27\lib\site-packages\Pythonwin
              C:\Python27\lib\site-packages\wx-2.8-msw-unicode
              C:\Users\tony\robotframework\rf_pi
```


and this is the cmd output:

[debug2.txt](https://github.com/nokia/RED/files/3223540/debug2.txt)

**Comment by kenifanying** [comment]
It's only occured when preference: 'Robot Framework -> Libraries -> Generate Python libraries libdocs in separate process' is disabled

**Comment by michalanglart** [comment]
What about python3? This may be related with the way python 2 handles unicode characters which your library prints in its constructor. By the way: why your library needs to log something in its constructor? Can you change the code so that it would log only when library is constructed during tests execution?

**Comment by michalanglart** [comment]
Regarding this one: `implement_array_function method already has a docstring` from your logs: googling for this phrase gives a bug in `numpy 1.16.3` when used with `python 2.7`

**Comment by kenifanying** [comment]
> 
> 
> What about python3? This may be related with the way python 2 handles unicode characters which your library prints in its constructor. By the way: why your library needs to log something in its constructor? Can you change the code so that it would log only when library is constructed during tests execution?

The library I use is python2 only , and it was written by other department colleagues. I know it's stupid, but i can't change it.

**Comment by michalanglart** [comment]
Well I don't know if there is something we can do - as a workaround please enable `Generate Python libraries libdocs in separate process` preferences.

Keep in mind that python2 end of life is 7 months away and sooner or later Robot Framework will probably drop support for this version.

**Comment by Mike2206** [comment]
Hi again,

Thats the last entry in my log:

!ENTRY org.eclipse.egit.ui 2 0 2019-04-27 13:14:32.216
!MESSAGE Warning: The environment variable HOME is not set. The following directory will be used to store the Git
user global configuration and to define the default location to store repositories: 'C:\Users\Michaś'. If this is
not correct please set the HOME environment variable and restart Eclipse. Otherwise Git for Windows and
EGit might behave differently since they see different configuration options.
This warning can be switched off on the Team > Git > Confirmations and Warnings preference page.
!SESSION 2019-04-27 13:15:02.773 -----------------------------------------------
eclipse.buildId=4.9.0.I20180906-0745
java.version=1.8.0_211
java.vendor=Oracle Corporation
BootLoader constants: OS=win32, ARCH=x86, WS=win32, NL=pl_PL
Framework arguments:  -product org.eclipse.epp.package.jee.product
Command-line arguments:  -os win32 -ws win32 -arch x86 -product org.eclipse.epp.package.jee.product

!ENTRY org.eclipse.egit.ui 2 0 2019-04-27 13:15:13.536
!MESSAGE Warning: The environment variable HOME is not set. The following directory will be used to store the Git
user global configuration and to define the default location to store repositories: 'C:\Users\Michaś'. If this is
not correct please set the HOME environment variable and restart Eclipse. Otherwise Git for Windows and
EGit might behave differently since they see different configuration options.
This warning can be switched off on the Team > Git > Confirmations and Warnings preference page.

!ENTRY org.eclipse.core.jobs 4 2 2019-04-27 13:16:24.992
!MESSAGE An internal error occurred during: "Looking for Python installations".
!STACK 0
org.rf.ide.core.environment.RobotCommandRpcExecutor$RobotCommandExecutorException: Unable to start XML-RPC server on file: C:\Python
	at org.rf.ide.core.environment.PythonInterpretersCommandExecutors.getRobotCommandExecutor(PythonInterpretersCommandExecutors.java:98)
	at org.rf.ide.core.environment.PythonInstallationDirectoryFinder$PythonInstallationDirectory.getRobotVersion(PythonInstallationDirectoryFinder.java:106)
	at org.robotframework.ide.eclipse.main.plugin.preferences.InstalledRobotsPreferencesPage.addOnlyNonExisting(InstalledRobotsPreferencesPage.java:150)
	at org.robotframework.ide.eclipse.main.plugin.preferences.InstalledRobotsPreferencesPage.access$7(InstalledRobotsPreferencesPage.java:145)
	at org.robotframework.ide.eclipse.main.plugin.preferences.InstalledRobotsPreferencesPage$1.run(InstalledRobotsPreferencesPage.java:134)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)


btw - is there a way to generate a new log? Apart from that changing the TEMP folder to a different location produces the same result

**Comment by michalanglart** [comment]
> is there a way to generate a new log?

What do you mean by that? Eclipse logs unhandled errors by writing to this file. If you delete it it will be created again.

If this is the last entry in your `.log` file then it looks like:
a) you haven't run into any unhandled error since 27th April
b) you switched the workspace to different location which has its own `.log` file but you're looking into the old one
c) ...

Let's just start again: what is the eclipse version you're currently using and which version of RED do you have installed? Can you reproduce the problem when starting eclipse/RED with fresh workspace? If so please provide us with the .log from this new workspace plus configuration path (how does your TMP/TEMP environment variables are set when problem is reproduced on fresh workspace?

**Comment by Mike2206** [comment]
Ok I have changed the workspace and there is still the same exception... Both events saved to fresh log which is here:
[log.log](https://github.com/nokia/RED/files/3243625/log.log)


I tried also to change the TEMP folder and it produces the same result, screenshots below
![Exception](https://user-images.githubusercontent.com/49831799/58746583-0d841e80-8460-11e9-8d13-af64be98f199.png)
![Exception2](https://user-images.githubusercontent.com/49831799/58746584-0d841e80-8460-11e9-8a5b-76384bd584dc.png)

Remember also I am a beginner, you need to tell me what to do step by step if possible

**Comment by michalanglart** [comment]
Can you please go to: `Help -> About Eclipse/RED -> Configuration` then copy the content of this tab into some file and attach it here?

**Comment by Mike2206** [comment]
Here you go:
[Config.txt](https://github.com/nokia/RED/files/3249038/Config.txt)

**Comment by michalanglart** [comment]
What happens in following scenario:
1. Open `RED`
2. Go to preference page `Robot Framework -> Installed frameworks`
3. Wait for exception
4. Write down the paths from exception (path to `python` and path to `robot_session_server.py`). In your screen 3 comments above those paths are: `C:\Python\python.exe` and `D:\TEMP\RobotTempDir8423404396545870128\robot_session_server.py`
5. Open `cmd` terminal
6. Launch following command using paths from above:
`c:\path\to\python\python.exe d:\path\to\robot_session_server.py 12345`

What happens? Does this script crash? What is the output?

**Comment by Mike2206** [comment]
Outputs are as follows:
![Python](https://user-images.githubusercontent.com/49831799/58906055-62978d00-870b-11e9-9ce6-35131dad62af.png)
![RobotSession](https://user-images.githubusercontent.com/49831799/58906057-63302380-870b-11e9-9d5e-96c7825d26f6.png)

**Comment by michalanglart** [comment]
Uhmm...

I asked you to perform single command:

> Launch following command using paths from above:
c:\path\to\python\python.exe d:\path\to\robot_session_server.py 12345

In your case this can be :
```
c:\
cd Python
python.exe D:\TEMP\RobotTempDir5770079580717458252\robot_session_server.py 12345
```

This should start session server scirpt on port 12345. In my case it looks like this (the paths are obviously different):

![cmd](https://user-images.githubusercontent.com/14932410/58944382-0885de80-8782-11e9-9392-8d65f4808f31.png)

**Comment by Mike2206** [comment]
Ok, so this is the resuult:

![Robot](https://user-images.githubusercontent.com/49831799/59150323-c37fd780-8a21-11e9-864c-b1a937953131.png)

**Comment by michalanglart** [comment]
Ok so it looks that you can start the server without any problems but RED is unable to do it itself. Can you then try to show the server ouput in RED as described here: http://nokia.github.io/RED/help/user_guide/diagnostics.html (paragraph 1.). Please show us the output.

**Comment by Mike2206** [comment]
This is what was printed in the console:

![Console](https://user-images.githubusercontent.com/49831799/59563581-38639a80-903c-11e9-8d6d-8136877bad14.png)

**Comment by michalanglart** [comment]
Ok, we have no clue currently to say why you are able to run session server from console but once RED is trying to run it it simple terminates immediately.

I've prepared small piece of code which is normally executed by RED in order to try to run the session server straight from java. Please download attached jar (inside the zip file) and try to run it this way:

```
java -jar serverRun.jar
```

It should first look for the path of your python interpreter and then it should run our session server script using found interpreter. Then it should connect to that server call a method on it and then kill the server. Please show us the output. In my case it looks like this:
![serverrun](https://user-images.githubusercontent.com/14932410/59678611-367d1100-91cd-11e9-9e39-e21031f99f1c.png)


[serverRun.zip](https://github.com/nokia/RED/files/3301280/serverRun.zip)

**Comment by 1348641743** [comment]
Hello, I also have the above problems. At present, just like the previous situation, you can start it manually directly. If you use red, an error will be reported. Now, I will use the file you gave to execute and report the following error. What should I do?
![image](https://user-images.githubusercontent.com/36215871/66895020-68d83700-f024-11e9-8c91-1fd6dcb05536.png)

**Comment by michalanglart** [comment]
Hi,

this does not tell us much unfortunately as of now. We'll prepare similar tool with logging added everywhere it is possible so that it would tell us more.

**Comment by 1348641743** [comment]
> 你好
> 
> 到目前为止，这并不能告诉我们很多。我们将准备类似的工具，并在可能的地方添加日志记录，以便它可以告诉我们更多信息。

Sorry, thanks for reply, this is a red original error logs and setup logs, want to help you, here I first started using the manual method of first use this local connection, you have here a good method, please contact me at, thank you very much
[installation.txt](https://github.com/nokia/RED/files/3733138/installation.txt)
[log.txt](https://github.com/nokia/RED/files/3733140/log.txt)

**Comment by modrz** [comment]
Hi,
please run following diagnostic tool and show the output.

`java -jar serverRun.jar`
[serverRun.zip](https://github.com/nokia/RED/files/3734331/serverRun.zip)

In my case it look like this:
![image](https://user-images.githubusercontent.com/22540835/66920264-3a0d9100-f023-11e9-976b-3facc9c96ddc.png)

**Comment by 1348641743** [comment]
Hello, this is the operation result. What's the problem?
![image](https://user-images.githubusercontent.com/36215871/66976833-f45dd080-f0d5-11e9-85df-ad37cd761d62.png)

**Comment by modrz** [comment]
Hi,

Thanks for feedback.

Now we know that the problem is connected to retrieving robot framework version which takes too much time in your machine.

I would like ot ask you to check again and send me outputs of following two commands using attached files.

```
python robot_version.py
java -jar serverRun.jar
```

[robot_version.zip](https://github.com/nokia/RED/files/3737487/robot_version.zip)
[serverRun.zip](https://github.com/nokia/RED/files/3737486/serverRun.zip)

**Comment by 1348641743** [comment]
OK, this is the result of the operation. It seems that we still haven't got it.
![image](https://user-images.githubusercontent.com/36215871/66981016-27f42700-f0e5-11e9-9aac-c4243549d02f.png)

**Comment by modrz** [comment]
Thanks, are you sure you used newest serverRun.jar?
In my case output looks like this (module import logs available):
![image](https://user-images.githubusercontent.com/22540835/66981530-50176200-f0b4-11e9-9362-cfb9109b135d.png)

**Comment by 1348641743** [comment]
Sorry, the file should be executed incorrectly. This is the correct execution result.
![image](https://user-images.githubusercontent.com/36215871/66981860-92a66200-f0e7-11e9-8b23-4ade49ec8c5c.png)

**Comment by modrz** [comment]
Thanks for cooperation,
I have extracted default 30s connection timeout to parameter now.
Please check the last thing for me (using newest attached serverRun.jar).
Provide bigger timeout (for example 2 minutes).

```
java -jar serverRun.jar 120
```
[serverRun.zip](https://github.com/nokia/RED/files/3737650/serverRun.zip)

**Comment by modrz** [comment]
I also hope it can be solved.
It looks like the line `import robot` somehow hangs python process and increasing timeout to 2 minutes does not solve it.
I have changed this replacing `import robot` to `import robot.version`, maybe it will help here.
Can you check it for me again?
[serverRun.zip](https://github.com/nokia/RED/files/3737750/serverRun.zip)

**Comment by modrz** [comment]
And one more thing, check also this to ensure that the same python is used for both checks:
`C:\Python27\python.exe robot_version.py`

**Comment by 1348641743** [comment]
OK, let me try again. 
This is the result. 
Is there any other good way to solve it?
![image](https://user-images.githubusercontent.com/36215871/66985880-50355300-f0f0-11e9-8217-c04290ccadf7.png)

**Comment by modrz** [comment]
How about this?
```
where python
C:\Python27\python.exe robot_version.py
```

Is it default Robot Framework installation?
Have you modified anything?

We don't know yet what causes such behavior, it is really strange.

**Comment by 1348641743** [comment]
The result is like this. I think windows has a python, which can't be deleted. I don't know if it has an impact on the result.I'll try to delete it and try again.
![image](https://user-images.githubusercontent.com/36215871/66987375-3e08e400-f0f3-11e9-86f0-824aaeacc879.png)

**Comment by 1348641743** [comment]
Delete or not. Distressed, do you mean it may be related to the robot framework?
![image](https://user-images.githubusercontent.com/36215871/66989136-5c241380-f0f6-11e9-8fea-6d9b39953d3d.png)

**Comment by 1348641743** [comment]
Congratulations, it seems normal. You are very good. Why is that? By the way, how can I modify it in red so that it can be used normally?
![image](https://user-images.githubusercontent.com/36215871/66990049-0d777900-f0f8-11e9-8c4d-aa7e9618cf9b.png)

**Comment by modrz** [comment]
Great news!

I am not sure what causes this problem.
It looks like importing robot and retrieving installed version before starting session server and thread responsible for shutting down this server solves the issue.

Before I prepare some solution to fix it also in RED please confirm that this also works:
[serverRun.zip](https://github.com/nokia/RED/files/3738140/serverRun.zip)

**Comment by 1348641743** [comment]
All right, are you packing again? The next version will be fine. I need to wait for this one, right?
![image](https://user-images.githubusercontent.com/36215871/66992749-e7a0a300-f0fc-11e9-846b-f95c8e12e88a.png)

**Comment by modrz** [comment]
You can check if following feature solves your problem with RPC server starting.

Just uninstall RED feature from your RED product in `Help -> About RED -> Installation Details -> Installed Software`.
Then after restart install feature from attachment in `Help -> Install New Software... -> Add... -> Archive...`.
After another restart RPC server should be started.
Let us know if this solution works.

[RED_feature_0.9.1.201910170949.zip](https://github.com/nokia/RED/files/3738628/RED_feature_0.9.1.201910170949.zip)

**Comment by 1348641743** [comment]
Uninstall this. After I uninstall it, I can't open red.
![image](https://user-images.githubusercontent.com/36215871/67059202-b5d61d80-f18a-11e9-8ac7-6191815d692d.png)

**Comment by 1348641743** [comment]
Oh, I know what's going on. I should delete the editor. After installation, it works normally. Thank you for your patience these days. Thank you.
![image](https://user-images.githubusercontent.com/36215871/67059758-d7380900-f18c-11e9-90c4-bbbd04a4d012.png)

**Comment by modrz** [comment]
Great news!
This fix will be released in next version.
Thanks for your cooperation.

**Comment by gvdwinkel** [comment]
All of a sudden while using Red to familiarize myself with Red + Robot Framework I stumbled upon errors everywhere in Red with "Unable to communicate with XML-RPC server".
![image](https://user-images.githubusercontent.com/57731052/68805959-2165ba80-0665-11ea-9ed8-512301551e9b.png)

I tried to update to your latest version here, but still same problem persisted.
It started to show up when I tried to show documentation, then it continues to show XML-RPC server errors almost everywhere. I'm not able to use my project anymore, even after restarting Red.

![image](https://user-images.githubusercontent.com/57731052/68806132-84efe800-0665-11ea-98b9-a966c425fcdd.png)

I was able to trace back where my problems are coming from.

I had added a Python file with following code:

stagingvars = {"BASEURL":"https://api-staging.something.com/network/things"}
prodvars = {"BASEURL":"https://api-prod.something.com/network/things"}
emptyvars = {"DUMMY":""}

def get_variables(env):

    if env == 'PROD':
        return prodvars
    elif env == 'STAGING':
        return stagingvars
    else: 
        return emptyvars
===

This code did not gave me any problems until I decided to use the upper function in this code:

**from numpy.core.defchararray import upper**
stagingvars = {"BASEURL":"https://api-staging.ioteconnect.com/network/things"}
prodvars = {"BASEURL":"https://api-prod.iotconnect.com/network/things"}
emptyvars = {"DUMMY":""}

def get_variables(env):

    if upper(env) == 'PROD':
        return prodvars
    elif env == 'STAGING':
        return stagingvars
    else: 
        return emptyvars

===

Now after some seconds Red starts to throw me the XML-RPC server errors everywhere, right until I remove the "from numpy.core.defchararray import upper" and upper function from the code.

1) I work under WIndows 7 64bit
2) I installed Python 3.7 (64bit)
3) pip install robotframework
3) pip install --upgrade RESTinstance
4) downloaded Red and unzipped it and run Red from the unzipped folder
5) created a project to familiarize myself with Red, Robot Framework and REST API calls tests

If I need to provide anything more, please let me know.

pip the Robot Framework

**Comment by modrz** [comment]
It looks like a different issue, I have created new one, lets continue discussion there.

**Comment by michalanglart** [comment]
The fix mentioned here: https://github.com/nokia/RED/issues/295#issuecomment-543519286 is now released in 0.9.1
